tests/test/scheduler/test_executor.cpp

```diff
@@ -330,6 +330,34 @@ TEST_CASE_METHOD(TestExecutorFixture,
     REQUIRE(restoreCount == 0);
 }
 
+TEST_CASE_METHOD(TestExecutorFixture,
+                 "Test executing function repeatedly and flushing",
+                 "[executor]")
+{
+    std::shared_ptr<BatchExecuteRequest> req =
+      faabric::util::batchExecFactory("dummy", "simple", 1);
+    uint32_t msgId = req->messages().at(0).id();
+    std::vector<std::string> actualHosts;
+
+    int numRepeats = 20;
+    for (int i = 0; i < numRepeats; i++) {
+        std::vector<std::string> actualHosts =
+          executeWithTestExecutor(req, false);
+        faabric::Message result =
+          sch.getFunctionResult(msgId, SHORT_TEST_TIMEOUT_MS);
+        std::string expected =
+          fmt::format("Simple function {} executed", msgId);
+        REQUIRE(result.outputdata() == expected);
+
+        // We sleep for the same timeout threads have, to force a race condition
+        // between the scheduler's flush and the thread's own cleanup timeout
+        SLEEP_MS(conf.boundTimeout);
```

Comment:
This could end up being quite a long sleep depending on the default config, and it's multiplied by 20, so it would be good to set this to something very short at the start of this test (e.g. 100ms).

Reply:
Good point, thanks